rfcs/plugin-api/experiment-2.ts

```diff
@@ -0,0 +1,45 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* eslint-disable @typescript-eslint/no-empty-function */
+import { Config, EnvironmentFunctions } from "@inlang/core/src/config"
+
+export const myPlugin = createPlugin<{ pathPattern: string }>(({ pluginConfig, env }) => {
+	return {
+		id: "samuelstroschein.plugin-json",
+		defineConfig: (config) => {
+			if (pluginConfig.pathPattern === undefined) {
+				throw new Error("pathPattern is required")
+			}
+			config.readResources = readResources({ pluginConfig, env })
+			config.languages = getLanguages({ pluginConfig, env })
+		},
+	}
+})
+
+/**
+ * The function to configure a plugin.
+ *
+ * @example
+ *   plugins: [
+ * 	 	myPlugin({
+ * 	   	pathPattern: "hello",
+ * 	 	})
+ *   ]
+ */
+type PluginConfigFunction<PluginConfig> = (
+	pluginConfig: PluginConfig,
```

Comment:
I would rename `pluginConfig` to `settings` or something other. so wen don't have to explain different `config`s.

Reply:
Hm, I find `pluginConfig` relatively straightforward. `settings` could mean settings of inlang (at least to me). 

@inlang/team please vote on `pluginConfig` or `settings`.

Reply:
Yes, and probably 
- `defineConfig` -> `config` 
and 
- `createPlugin` -> `plugin` 
if they are no naming clashes with this approach.

Reply:
@inlang/team please vote on `pluginConfig` or `settings`, or `options`.

Reply:
Okay, going for `settings`.